### PR TITLE
New Perl

### PR DESCRIPTION
**Pull request by fosslinux**
This has taken me a very very long time due to many complexities and interruptions such as exams but we now have the latest perl.

This has a number of random changes and new software, including:

- bison 2.3, backwards bootstraped from 3.4.1, needed to compile perl 5.10
- perl 5.10, used in bootstrap of 5.32.1
- dist, perl's 'Configure' creator
- Newer, less buggy make needed for perl
- Bison 3.4.2 which is built normally and is less buggy needed for perl
- Finally perl 5.32.1, a suprisingly massive jump that happened to work

This adds 10-15 mins to the bootstrap.

### CODE REVIEW

File: sysa.py [hunk 1]
```diff
@@ -441,3 +441,7 @@ def get_packages(self):
 
         # mpc 1.2.1
         self.get_file("https://mirrors.kernel.org/gnu/mpc/mpc-1.2.1.tar.gz")
+
+        # bison 2.3
+        self.get_file(["https://mirrors.kernel.org/gnu/bison/bison-2.3.tar.bz2",
+                      "https://git.savannah.gnu.org/cgit/gnulib.git/snapshot/gnulib-b28236b.tar.gz"])
```
Comment (stikonas) [inline sysa.py:450]: all other imports align at quotation mark, so maybe a add a space.

File: sysa/bison-3.4.2/bison-3.4.2.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# SPDX-FileCopyrightText: 2021 fosslinux <fosslinux@aussies.space>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+src_prepare() {
+    default_src_prepare
```
Comment (stikonas) [inline sysa/bison-3.4.2/bison-3.4.2.sh:6]: `default`

File: sysa/bison-3.4.2/bison-3.4.2.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# SPDX-FileCopyrightText: 2021 fosslinux <fosslinux@aussies.space>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+src_prepare() {
```
Comment (stikonas) [inline sysa/bison-3.4.2/bison-3.4.2.sh:5]: remove `.info` files.

File: sysa/perl-5.32.1/checksums [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+f4bd302f99c650344c83973867a0fbd2ce3bedb35460c99f964add78cfda7620  /after/lib/perl5/5.32.1/x86_64-linux/auto/Digest/SHA/SHA.a
+1adda90e944ad90f66a6d6beeb73f36eb15346868f86fcf2428e9468e0df924c  /after/lib/perl5/5.32.1/x86_64-linux/auto/Digest/MD5/MD5.a
+913b4c30b4444bb5d20fdfb1235b0bad897900247d6c90f20453f4b3fff88d22  /after/lib/perl5/5.32.1/x86_64-linux/auto/Devel/Peek/Peek.a
+40bd3da520ef0b12afb7a6e422bba60618a294ea48315a0c3788b238ceec1cff  /after/lib/perl5/5.32.1/x86_64-linux/auto/Data/Dumper/Dumper.a
+11a5a396f223fca1c0f663f92e2ccca804c0e1bcc8ebdc7aa6c02007c0ef6eb2  /after/lib/perl5/5.32.1/x86_64-linux/auto/Cwd/Cwd.a
+cbace9433582cb7a952f8765351e5e8ac24c3d08c5cc659fc57fc5d001f85a46  /after/lib/perl5/5.32.1/x86_64-linux/auto/Compress/Raw/Zlib/Zlib.a
+97297837e713b3c32fdf15d3beb4df27dd04d36770520a39cb9609bfe03e8b95  /after/lib/perl5/5.32.1/x86_64-linux/auto/Compress/Raw/Bzip2/Bzip2.a
+a00d9d1d16cb84df222e42726962751d6296aa4ef5b43d3e3d39b5db677524be  /after/lib/perl5/5.32.1/x86_64-linux/auto/B/B.a
+b9aa5df8e7790d25b610d3568a1fb5ec19d4b7a24ea89032b3e48e6baeff7829  /after/lib/perl5/5.32.1/x86_64-linux/auto/IO/IO.a
```
Comment (stikonas) [inline sysa/perl-5.32.1/checksums:82]: That will cause problems on `i386` kernel.

File: sysa/perl-5.32.1/perl-5.32.1.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# SPDX-FileCopyrightText: 2021 Andrius Štikonas <andrius@stikonas.eu>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+src_prepare() {
+    default_src_prepare
```
Comment (stikonas) [inline sysa/perl-5.32.1/perl-5.32.1.sh:6]: `default`

File: sysa/perl-5.32.1/perl-5.32.1.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+src_configure() {
+    ./Configure -des -Dprefix="${PREFIX}" -Dcc=gcc -Dusedl=false -Ddate=':' -Dccflags="-U__DATE__ -U__TIME__"
+}
+
+src_install() {
+    # Remove old perl
+    rm -rf "${PREFIX}"/lib/perl5/
+
+    default_src_install
```
Comment (stikonas) [inline sysa/perl-5.32.1/perl-5.32.1.sh:39]: `default`

File: sysa/perl-5.32.1/perl-5.32.1.sh [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+    # Regenerate configure + config_h.SH
+    rm Configure config_h.SH
+    ln -s ../perl-5f2dc80/regen-configure/.package .
+    ln -s ../perl-5f2dc80/regen-configure/U .
+    metaconfig -m
+}
+
+src_configure() {
+    ./Configure -des -Dprefix="${PREFIX}" -Dcc=gcc -Dusedl=false -Ddate=':' -Dccflags="-U__DATE__ -U__TIME__"
```
Comment (stikonas) [inline sysa/perl-5.32.1/perl-5.32.1.sh:32]: Maybe `-Dtargetarch` will help to keep it 32-bit for now?

**Review by stikonas — CHANGES_REQUESTED** [review]

File: sysa/dist-3.5-236/patches/env.patch [hunk 8]
```diff
@@ -0,0 +1,17 @@
+This is not valid Perl, I don't really understand why it is here anyway.
+
+SPDX-FileCopyrightText: 2021 fosslinux <fosslinux@aussies.space>
+
+SPDX-License-Identifier: Artistic-1.0
```
Comment (stikonas) [inline sysa/dist-3.5-236/patches/env.patch:5]: This causes `reuse` linter issues.

**Review by stikonas — CHANGES_REQUESTED** [review]

File: SHA256SUMS.sources [hunk 9]
```diff
@@ -61,3 +61,12 @@ c6c37e888b136ccefab903c51149f4b7bd659d69d4aea21245f61053a57aa60a  tar-1.12.tar.g
 de23af78fca90ce32dff2dd45b3432b2334740bb9bb7b05bf60fdbfc396ceb9c  tcc-0.9.27.tar.bz2
 1b700d375e253e88bdf79caa244284a899b80f59fd64fdd3aaa6408b7bd715ae  untar.c
 166c48d2842519bc4f96333bff9e265f8cdda44d38e40594ef3f9bbb52890490  xz-5.0.5.tar.bz2
+b10d7e9e354be72aee4e4911cf19dd27b5c527d4e7200857365b5fcdeea0dffb  bison-2.3.tar.bz2
+0190f28cb155fedd22bf8558c3e8705eed9eacfb7ae29e7508d025a68eb90899  gnulib-b28236b.tar.gz
+9385f2c8c2ca8b1dc4a7c31903f1f8dc8f2ba867dc2a9e5c93012ed6b564e826  perl-5.10.1.tar.bz2
+a3fd3de1f1684f689d823af367ba18bba9fb000d4cef94de60eeb63b1dfe2a6c  dist-3.5-236.tar.gz
```
Comment (stikonas) [inline SHA256SUMS.sources:67]: Hmm, I'm getting checksum mismatch with this file. Any idea what is going wrong?

Comment (fosslinux) [reply SHA256SUMS.sources:67]: I am unsure. Perhaps my local version has corruption?

File: sysa/perl-5.10.1/patches/undefined_hack.patch [hunk 10]
```diff
@@ -0,0 +1,124 @@
+SPDX-FileCopyrightText: 2021 fosslinux <fosslinux@aussies.space>
+
+SPDX-License-Identifier: Artistic-1.0-Perl
+
+There are a bunch of aliases that perl's build system somehow makes but
+I'm not sure how to appropriately do them, so just patch them.
```
[the remark this answers is not in the document]
Comment (fosslinux) [reply sysa/perl-5.10.1/patches/undefined_hack.patch:6]: How do you mean? These were all encountered in the build process.

File: sysa/bison-2.3/bison-2.3.sh [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+src_prepare() {
+    default_src_prepare
+
+    ../../import-gnulib.sh
+
+    AUTOPOINT=true autoreconf-2.65 -fi
+
+    # Remove pregenerated files
+    rm src/parse-gram.c src/parse-gram.h
```
[the remark this answers is not in the document]
Comment (fosslinux) [reply sysa/bison-2.3/bison-2.3.sh:13]: Oops, how did I miss those!

File: SHA256SUMS.sources (hunk 9 above)
Comment (fosslinux) [reply SHA256SUMS.sources:67]: I'm pretty sure this case. Please post checksum. New checksum seems to work?

Comment (stikonas) [reply SHA256SUMS.sources:67]: 54437ae4211867de7ad55723f68b94c29cb2f08a23f431e0bbbc9f34ee384f47  dist-3.5-236.tar.gz

**Comment by fosslinux** [comment]
Updated. I think everything has been addressed. Also fixed make `--build=` option (force i686).

File: sysa/dist-3.5-236/dist-3.5-236.sh [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    cd ..
+
+    cd kit
+    ./manifake.SH
+    cd ..
+}
+
+src_install() {
+    install mcon/metaconfig "${PREFIX}/bin/"
+    install  kit/manifake "${PREFIX}/bin/"
```
Comment (stikonas) [inline sysa/dist-3.5-236/dist-3.5-236.sh:27]: double space

**Review by stikonas — APPROVED** [review]

**Pull request merged**